library/java/io/envoyproxy/envoymobile/engine/types/EnvoyKeyValueStore.java

```diff
@@ -0,0 +1,27 @@
+package io.envoyproxy.envoymobile.engine.types;
+
+public interface EnvoyKeyValueStore {
+  /**
+   * Read a value from the key value store implementation.
+   *
+   * @param key,     key identifying the value to be returned.
+   * @return String, value mapped to the key, or null if not present.
+   */
+  String read(String key);
+
+  /**
+   * Remove a value from the key value store implementation.
+   *
+   * @param key,     key identifying the value to be removed.
+   * @return String, value mapped to the key, or null if not present.
+   */
+  void remove(String key);
+
+  /**
+   * Save a value to the key value store implementation.
+   *
+   * @param key,     key identifying the value to be saved.
+   * @param value,   the value to be saved.
```

Comment:
nit: Looks like this returns void?

Reply:
Sorry I may be being dense - what's the fix here? I fixed the `@return` for remove, but this method doesn't have one.

Reply:
Oh, shoot. You're right. I somehow totally misread this. :(

Reply:
All good - thanks for all the comments. :)